src/demo/spawn.ts

```diff
@@ -0,0 +1,283 @@
+        return fin.Window.wrapSync(identity);
+    } else {
+        // Create the window ourselves
+        return createChildWindow(options);
+    }
+}
+
+async function createApplication(options: Omit<AppData, 'parent'>): Promise<Application> {
+    const uuid: string = options.id || `App-${Math.random().toString().substr(2, 4)}`;
+    const url = getUrl(options);
+    const position = getWindowPosition(options);
+    const size = getWindowSize(options);
+    const state = options.state || 'normal';
+
+    if (options.type === 'programmatic') {
+        const data: ApplicationOption = {
+            uuid,
+            name: uuid,
+            mainWindowOptions: {
+                ...position,
+                url,
+                frame: options.frame === undefined ? true : options.frame,
+                state,
+                autoShow: true,
+                saveWindowState: false,
+                defaultWidth: size.x,
+                defaultHeight: size.y
+            }
+        };
+        return startApp(fin.Application.create(data));
+    } else {
+        const queryOptions: Dictionary<string|number|boolean> = {
+            ...position as Required<typeof position>,
+            uuid,
+            url,
+            state,
+            defaultWidth: size.x,
+            defaultHeight: size.y,
+            frame: options.frame === undefined ? true : options.frame,
+            realmName: options.realm || '',
+            enableMesh: options.enableMesh || false,
+            runtime: options.runtime || await fin.System.getVersion(),
+            useService: options.useService !== undefined ? options.useService : false,
+            provider: options.provider || 'local'
+        };
+
+        const manifest = `http://localhost:3922/manifest?${
+            Object.keys(queryOptions)
+                .map(key => {
+                    return `${key}=${encodeURIComponent(queryOptions[key].toString())}`;
+                })
+                .join('&')}`;
+
+        return startApp(fin.Application.createFromManifest(manifest));
+    }
+}
+
+async function createChildWindow(data: Omit<WindowData, 'parent'>): Promise<_Window> {
+    const name: string = data.id || `Win-${Math.random().toString().substr(2, 4)}`;
+    const url = getUrl(data);
+    const position = getWindowPosition(data);
+    const size = getWindowSize(data);
+
+    const options:
+        WindowOption = {...position, name, url, frame: data.frame, autoShow: true, saveWindowState: false, defaultWidth: size.x, defaultHeight: size.y};
+    return fin.Window.create(options);
+}
+
+async function startApp(appPromise: Promise<Application>): Promise<Application> {
+    const app = await appPromise;
+    await app.run();
+    return app;
+}
+
+function getUrl(options: WindowData): string {
+    // Create URL from base URL + queryArgs
+    let url = options.url || 'http://localhost:3922/test/test-app.html';
```

Comment:
That's a good question which I don't really have an answer to. The default value needed changing because it pointed to some layouts demo window which doesn't exist. Having the test window as the default also means I don't have to type in a url every single time I call the functions from tests (which is the only place we're likely to be calling them).

Something to keep in mind is that this will be webpacked which would likely cause some difficulty if it was in /test/... as we treat all code in there as test utils running in node.